### ISSUE TITLE
Extension is excruciatingly slow at linting suddenly

### ISSUE THREAD

**Issue by acoutts** [Microsoft/vscode-typescript-tslint-plugin#83]
- tslint Version: 5.14.0
- typescript Version: 3.3.3

Steps to Reproduce:
1. Type out an export name that is not currently imported.
2. After 20-45 seconds, it will finally find the export and allow you to press tab to automatically import the required piece.

This has always been very fast but lately it has suddenly become so slow it's almost unbearable. I've sat waiting up to 45 seconds for it to finish "thinking" which usually takes no more than a few seconds.

I have a 5.0 GHz i7 and 32gb of RAM so the system resources are not an issue. When it's stuck lagging, there's a single thread of `tsserver.js` using 100% cpu but otherwise the system seems relaxed.

**Comment by egamma** [comment]
@acoutts no need to resubmit, github now supports to transfer issues to other repos

**Comment by mjbvz** [comment]
@acoutts Does this only happen when this extension is installed?

**Comment by acoutts** [comment]
> @acoutts Does this only happen when this extension is installed?

Yeah, since this is actually an issue in the extension itself as the tslint process doesn't run while using vscode otherwise.

**Comment by mjbvz** [comment]
Auto imports are provided by typescript itself, not by this plugin. This plugin could be slowing down normal typescript however. If you disable this extension and open the same project in VS Code, do automatic imports and errors show up more quickly?

Also can you share your project?

**Comment by acoutts** [comment]
Ah very good to know. Unfortunately I can't share the code as it is internal to the project, but I did test with the extension disabled and it still seems slow. In that case that means this is probably outside of the scope for this repository. Do you have any recommendations where to go next?
